src/formats/roam-json.ts

```diff
@@ -255,8 +315,39 @@ export class RoamJSONImporter extends FormatImporter {
 		return blockText;
 	};
 
-	private async jsonToMarkdown(graphFolder: string, attachmentsFolder: string, json: RoamPage | RoamBlock, indent: string = '', isChild: boolean = false): Promise<string> {
+	// setup to hold the newest and oldest timestamp value from a given page
+	newestTimestamp: number = 0;
+	oldestTimestamp: number = 0;
+
+	private async jsonToMarkdown(graphFolder: string, attachmentsFolder: string, json: RoamPage | RoamBlock, indent: string = '', isChild: boolean = false, checkYAMLoptions : boolean = false, runYAMLoptions: boolean = true, setTitleProperty: string, createdTimestamp: number, updatedTimestamp: number): Promise<string> {
```

Comment:
The `checkYAMLoptions` and `runYAMLoptions` parameters are confusing and seem redundant.

`runYAMLoptions` seems to just be `!isChild`, so I think we can remove that one.

I'm not clear on why `checkYAMLoptions` is necessary since this function seems to be changing it's value based on global config anyway.

Can we remove both of these?

Reply:
Probably (at least getting passed as parameters), and it's mostly due to my inexperience. The idea was to set `runYAMLoptions` up to run once per note/page (or like you said, `!isChild`), and check against the user picking one or more YAML options (currently for `title`, and the `created`/`updated` choices). That's what the switch statement (near line ~365ish) is doing:

```
switch (!checkYAMLoptions) {
	case this.fileDateYAML:
	case this.titleYAML:
	// add other options here
	// case this.{option-name}:
		checkYAMLoptions = true;
}
```

I'll see if I can rework all of that under `jsonToMarkdown` and make it clearer (or remove as much as possible)

Reply:
Ok, I think I got it working better now. I rewrote the switch and removed both `checkYAMLoptions` and `runYAMLoptions` from the `jsonToMarkdown` function - is it better as an if (since the default is set to `false`) or a ternary? I have it as an `if` at the moment:

```
if (this.fileDateYAML || this.titleYAML) {
	checkYAMLoptions = true;
}
```
vs
```
checkYAMLoptions = (this.fileDateYAML || this.titleYAML) 
	? true 
	: false;
```

and then updated this statement to remove `runYAMLoptions`:

```
if (checkYAMLoptions && !isChild) {
```

Reply:
To directly answer your question, you don't need an if or ternary. It can just be:

```
checkYAMLoptions = this.fileDataYAML || this.titleYAML
```

However, see my comment below. I think we can remove `checkYAMLoptions` completely and use these two dependent vars directly.